Additional Preprocessing for County Health data: train/test split, state maps, training data table(s)

frederick-douglas-pearce opened the pull request:
This is an extension to the work in the previous PR from the feature_data_preprocessing branch, specifically items 8-11 below:
1. [X] Remove 6793 duplicate rows
2. [X] Remove 4839 county or state level aggregate rows, i.e. rows with county_code = 0
3. [X] Remove 2 rows with null county value that can be identified (state is null but has state_code for 'AK')
4. [X] Remove 78 rows with null measure_name, measure_id, or year_span values
5. [X] Remove 13404 rows with null raw_value values
6. [X] Validate final table at correct grain: state, county, year_span, measure_name
  - Final table has 52 distinct state values and 12 distinct measure_name values
  - Note that not all county values actually refer to a county in a state (e.g. state='MD', county='Baltimore city') but I think we can ignore this for now
7. [X] Pivot data: one row per state-county pair, one column per measure-year pair
8. [X] Split data: create separate training and test data sets, only use training data from here on...
9. [X] State maps of training and test counts with geopandas
10. [ ] Filter the training data to year-specific dataframe(s) with all relevant measures
11. [ ] Write training data to postgres table in 'train' schema

Xuehong-pdx commented:
HI Fred,

I took a look at the final pivot table you made and not sure what your plan was to use these data for making predictions.  From my perspective, I am not sure how to use that table to make predictions.  so I played with it a little and transformed state, county and measure_name into categorical variables then made dummy variables for them.  Based on this data set, I then split them into train and test sets and tried to used knn to make predictions.  I had to randomly select only 50K rows to do the fitting since it took too long, but I was able to fit and make predictions.  The RMSE for the training set was 884 while it was 922 for the test set.   I used 3 neighbors based on grid search and the best score was 0.9.  I thought it was reasonable.  

I haven't thought through how to incorporating geopandas in this process, but thought I should let you know where I am at.

frederick-douglas-pearce commented:
I just added a commit with some minor changes to documentation. I'm going to merge this in tonight, then I'll create a new PR for the year_span parsing into a new year column. That next PR should get us to a place where I have cleaned up training and test data sets...

frederick-douglas-pearce commented:
Maybe you can create a new branch/PR with the additional analysis you performed? Up to you, but I can't really comment on it if I don't see the notebook

Xuehong-pdx commented:
I think I need to wait for you to push it up before I can create a PR.  My
merge request was rejected.

On Wed, Oct 21, 2020 at 4:05 PM Fred Pearce <notifications@github.com>
wrote:

> Maybe you can create a new branch/PR with the additional analysis you
> performed? Up to you, but I can't really comment on it if I don't see the
> notebook
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/Xuehong-pdx/Collaborative-Projects/pull/7#issuecomment-713938247>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJ64YMD4DQ5RONUQAHRYFELSL5SMHANCNFSM4RXK26FQ>
> .
>

frederick-douglas-pearce commented:
I just merged this one in. You can add your analysis to this notebook; however, I was hoping to keep this one focused on preprocessing and I still need to add the year column work. You can wait for me to do that first, or you can just add your notebook with a different name? Not sure what is best at this point so will leave it up to you

Xuehong-pdx commented:
I can add it with a different name and we can delete it afterwards if we
don't want to keep it.  I like to keep the space as clean as possible :-)

On Wed, Oct 21, 2020 at 5:48 PM Fred Pearce <notifications@github.com>
wrote:

> I just merged this one in. You can add your analysis to this notebook;
> however, I was hoping to keep this one focused on preprocessing and I still
> need to add the year column work. You can wait for me to do that first, or
> you can just add your notebook with a different name? Not sure what is best
> at this point so will leave it up to you
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/Xuehong-pdx/Collaborative-Projects/pull/7#issuecomment-714081047>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJ64YMEOZIE2J2K6SFYI2HTSL56ONANCNFSM4RXK26FQ>
> .
>
